### PR TITLE
Merge/nex 778/notfication object

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
Notification object moved the directory up.
The interface has required PHP docs.

Merge is a part of changes made in branch `feature/NEX-137/add-spanner-driver`

https://oat-sa.atlassian.net/browse/NEX-778

### CODE REVIEW

File: models/classes/notification/Notification.php [hunk 1]
```diff
@@ -19,9 +19,7 @@
  *
  */
 
-namespace oat\tao\model\notification\implementation;
```
Comment (boajer) [inline models/classes/notification/Notification.php]: Why you are changing location of this class?

Comment (bartlomiejmarszal) [reply models/classes/notification/Notification.php]: implementation directory should serve only different persistence implementation and notification is only dummy DTO object that implements `NotificationInterface`. I would keep it next to the interface which it's implementing to keep consistency in implement directory.

Comment (mike-ionut-mihai-sandu-tao) [reply models/classes/notification/Notification.php]: so according to our discussion , let's please keep this inside the implementation folder so we comply with the existing status quo,  and please extend this thus allowing you to use the functionality into the newsql implementation properly

Comment (jbout) [reply models/classes/notification/Notification.php]: I agree on a conceptual level that Notification is not part of the implementation, it is shared between implementations and should be on the lower level. But if we want to move it we should consider backward compatibility

**Comment by jbout** [comment]
By using the Notification as a shared object we no longer need the interface. Also the current NotificationServiceInterface should rely on the Notification object, current return types are incorrect

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: test/unit/models/classes/notification/NotificationTest.php [hunk 2]
```diff
@@ -0,0 +1,163 @@
+<?php declare(strict_types=1);
```
Comment (bartlomiejmarszal) [inline test/unit/models/classes/notification/NotificationTest.php]:
```suggestion
<?php
```

File: models/classes/notification/Notification.php [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA;
```
Comment (boajer) [inline models/classes/notification/Notification.php]: 2017-2020? Or how should it looks like?

Comment (bartlomiejmarszal) [reply models/classes/notification/Notification.php]: I would say 2020

Comment (bartlomiejmarszal) [reply models/classes/notification/Notification.php]:
```suggestion
 * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
```

**Pull request merged**